src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int length;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = tail = new Node<>(null, value, null);
+        } else {
+            tail = tail.next = new Node<>(tail, value, null);
+        }
+        length++;
     }
 
     @Override
     public void add(T value, int index) {
+        isIndexValidForAdd(index);
```

Comment:
you don't need to check the index for validity because you do it in the getNode() method